pacehrh/R/pace_cadre_process.R

```diff
@@ -0,0 +1,283 @@
+  # A more elaborate version of this computation is done for seasonality
+  # processing, but here we assume that (1) overhead values apply to the entire
+  # date range, and (2) overhead values aren't affected by month, so the
+  # seasonality curve is flat.
+
+  curve <- c(1,1,1,1,1,1,1,1,1,1,1,1) / 12
+
+  monthlyValuesMatrix <- matrix(data = 0, nrow = nrow(roles), ncol = (length(BVE$years) * length(curve)))
+
+  for (i in seq_along(roles$ScenarioID)) {
+    annual <- annualValuesMatrix[i,]
+    monthly <- as.vector(matrix(data = curve, ncol = 1) %*% matrix(data = annual, nrow = 1))
+    monthlyValuesMatrix[i,] <- monthly
+  }
+
+  monthStrs <- as.character(1:12)
+  s <- sapply(as.character(BVE$years), function(year) {
+    paste0(year, ".", monthStrs)
+  })
+
+  dimnames(monthlyValuesMatrix) <- list("Role" = roles$RoleID, "Month" = as.vector(s))
+
+  return(list(annualOverheads = annualValuesMatrix, monthlyOverheads = monthlyValuesMatrix))
+}
+
+#' Compute And Save Suite Results In Extended Format
+#'
+#' @param results Results structure (as returned by [RunExperiments()]).
+#' @param filepath Destination CSV file to write results. Default = NULL.
+#' @param run Name of experiment run
+#'
+#' @return Data frame version of CSV file.
+#' @export
+#'
+#' @md
+#' @examples
+#' \dontrun{
+#' }
+
+SaveExtendedSuiteResults <- function(results = NULL, filepath = NULL, run = "Run-1") {
+  if (is.null(results)) {
+    return(NULL)
+  }
+
+  trialIds <- names(results)
+  scenario <- BVE$scenario$UniqueID
+
+  # Create the usual output CSV file format
+  l <- lapply(seq_along(trialIds), function(index) {
+    r <- results[[index]]
+    if ("SeasonalityResults" %in% names(r)) {
+      return(.saveSuiteResults2(r, scenario, index, run))
+    } else {
+      return(.saveSuiteResults1(r, scenario, index, run))
+    }
+  })
+
+  out <- data.table::rbindlist(l)
+
+  # Convert Service_time from minutes to hours
+  out$Service_time <- out$Service_time / 60.0
+
+  # Grab the Tasks table
+  tasks <- BVE$taskData
+  data.table::setDT(tasks)
+
+  # Add a total number-of-contacts value to the Tasks table
+  tasks$NumContactsAnnual[is.na(tasks$NumContactsAnnual)] <- 0
+  tasks$NumContactsPerUnit[is.na(tasks$NumContactsPerUnit)] <- 0
+  tasks$NumContactsPer <-
+    tasks$NumContactsPerUnit + tasks$NumContactsAnnual
+
+  # Filter the Tasks table column set
+  tasks <-
+    tasks[, .(Indicator,
+              CommonName,
+              ClinicalOrNon,
+              ClinicalCat,
+              ServiceCat,
+              NumContactsPer)]
+
+  # Join Scenario information to the output
+  out <- dplyr::left_join(out,
+                          BVE$scenario[c("UniqueID",
+                                         "WeeksPerYr",
+                                         "HrsPerWeek",
+                                         "BaselinePop",
+                                         "DeliveryModel")],
+                          by = dplyr::join_by(Scenario_ID == UniqueID))
```

Comment:
We should add this to force user to upgrade to the minimum version that has join_by
```
Imports:
    dplyr (>= 1.0.0),
```
@rhanIDM had encounter the issue which is as described as here https://r-pkgs.org/description.html#sec-description-imports-suggests-minium-version

 _If someone has an older package that doesn’t have a function your package needs, they’ll get an unhelpful error message if your package does not advertise the minimum version it needs. However, if you state a minimum version, they’ll automatically get an upgrade when they install your package._

Reply:
Added minimum version check for dplyr.

Reply:
Added minimum version check for dplyr.